sgl-kernel/tests/test_lightning_attention_decode.py

```diff
@@ -0,0 +1,104 @@
+import torch
+from sgl_kernel import lightning_attention_decode
+
+
+def naive_lightning_attention_decode(q, k, v, past_kv, slope):
+    """Naive implementation of lightning attention decode"""
+    original_dtype = q.dtype
+    ratio = torch.exp(-slope)  # [h, 1, 1]
+
+    kv = past_kv
+    b, h, n, d = q.shape
+
+    output = []
+    for i in range(n):
+        kv = ratio * kv.to(torch.float32) + torch.einsum(
+            "... n d, ... n e -> ... d e",
+            k[:, :, i : i + 1],
+            v[:, :, i : i + 1],
+        )
+        qkv = torch.einsum(
+            "... n e, ... e d -> ... n d",
+            q[:, :, i : i + 1].to(torch.float32),
+            kv.to(torch.float32),
+        )
+        output.append(qkv)
+    output = torch.concat(output, dim=-2)
+
+    return output.to(original_dtype), kv
+
+
+def test_lightning_attention_decode():
+    if not torch.cuda.is_available():
+        return
+
+    device = torch.device("cuda")
+    configs = [
```

Comment:
Use pytest instead

Reply:
Solved all comments in [commit](https://github.com/sgl-project/sglang/pull/3030/commits/cf872fa5d351e4ecc18955c7e4cf7d501af8a31b).